### PR TITLE
prov/efa: introduce FI_EFA_IFACE to restrict visible NICs

### PR DESCRIPTION
**Pull request by wenduwan**
This patch introduces an environment variable to limit which NICs are visible to the application via fi_getinfo.

### CODE REVIEW

**Comment by j-xiong** [comment]
Could it be named FI_EFA_IFACE to be consistent with other providers?

**Comment by shijin-aws** [comment]
> Could it be named FI_EFA_IFACE to be consistent with other providers?

Could you share a link to the implementation of other providers?

**Comment by j-xiong** [comment]
@shijin-aws See these man pages about **FI_TCP_IFACE** and **FI_VERBS_IFACE**:
https://ofiwg.github.io/libfabric/main/man/fi_tcp.7.html
https://ofiwg.github.io/libfabric/main/man/fi_verbs.7.html

**Comment by wenduwan** [comment]
@j-xiong Thank you! PR is updated accordingly.

**Comment by wenduwan** [comment]
@zachdworkin I imagine the intel CI failure is unrelated, correct?

**Comment by shijin-aws** [comment]
Should we advertise this env in man page?

**Comment by j-xiong** [comment]
Yes, I think so.

File: prov/efa/src/efa_env.h [hunk 1]
```diff
@@ -92,7 +95,38 @@ static inline bool efa_env_has_use_device_rdma() {
...
+
+	if (!strncmp("all", efa_env.iface, 3))
+		return true;
+
+	while (match < end) {
+		match = strstr(match, name);
+		if (!match)
+			return false;
+
+		if ((match > efa_env.iface && *(match - 1) != ',') ||
```
Comment (shijin-aws) [inline prov/efa/src/efa_env.h:119]: hmmm how do we compare two strings here?

**Comment by wenduwan** [comment]
> Should we advertise this env in man page?

Good point. Added documentation.

File: prov/efa/src/efa_env.h [hunk 2]
```diff
@@ -92,7 +95,38 @@ static inline bool efa_env_has_use_device_rdma() {
...
+		    ((match + strlen(name)) < end && *(match + strlen(name)) != ',')) {
+			/* Skip partial match */
+			match += strlen(name);
```
Comment (shijin-aws) [inline prov/efa/src/efa_env.h:122]: Clarified with @wenduwan offline. But this logic is not intuitive to new readers.

**Review by shijin-aws — APPROVED** [review]

**Comment by shijin-aws** [comment]
bot:aws:retest

**Pull request merged**